packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerCache.java

```diff
@@ -34,42 +33,48 @@
       "flutter_image_picker_pending_image_uri";
   private static final String SHARED_PREFERENCES_NAME = "flutter_image_picker_shared_preference";
 
-  private static SharedPreferences getFilePref;
+  private SharedPreferences prefs;
```

Comment:
I see all the null check are removed. I am not a Java expert but how do we ensure the `prefs` is not null to prevent NPE?

Reply:
.`static` methods and variables are associated with the class they are declared.
Non static methods and variables are associated with an instance of an object of a given type.

When I removed `static` declaration for `prefs` to make it part of an instance of `ImagePickerCache`. To access `prefs`, I need an instance of `ImagePickerCache`.
So we know `prefs` is never null because it is being instantiated in the constructor of  `ImagePickerCache`.

As `prefs` is no longer `static`, the `static` methods were converted to instance methods as well. Static methods can't access non static variables/methods after all.

As a rule of thumb: Avoid static methods/variables. They are hard to write unit tests and they are memory leak and error prone.